src/sentry/api/event_search.py

```diff
@@ -531,7 +532,9 @@ def visit_specific_time_filter(self, node, children):
             return self._handle_basic_filter(search_key, "=", SearchValue(date_value))
 
         try:
-            from_val, to_val = parse_datetime_value(date_value)
+            from_val, to_val = parse_datetime_value(
+                date_value, exact_timestamp=self.exact_timestamp
+            )
```

Comment:
Instead of adding a modal parameter, should we have a separate function that creates a single value instead of a range?

Reply:
I don't think so because the user can still pass in just the date without the time, and in that case, I'd expect the entire day rather than just at midnight. And in that case, I'd be duplicating all the logic here except the last bit.